onnxruntime/core/providers/cpu/tensor/scatter.cc

```diff
@@ -32,16 +50,31 @@ ONNX_CPU_OPERATOR_KERNEL(
 
 template <class Tin, class Tdata>
 Status CopyScatterData(const Tensor* data_input, const Tensor* indices_input, const Tensor* updates_input,
-                       const int64_t axis, Tensor* data_output) {
+                       const int64_t axis, Tensor* data_output, bool is_scatter_elements) {
   const TensorShape& input_data_shape = data_input->Shape();
-  const Tin* indices_data = indices_input->template Data<Tin>();
+  const Tin* indices_data_raw= indices_input->template Data<Tin>();
   const auto num_indices = indices_input->Shape().Size();
+
+  std::vector<Tin> indices_data;
+  indices_data.reserve(num_indices);
+
+  auto axis_dim_limit = input_data_shape[axis];
+
   for (int64_t i = 0; i < num_indices; ++i) {
-    Tin idx = indices_data[i];
-    if (idx < 0 || idx >= input_data_shape[axis]) {
-      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "indices element out of data bounds, idx=", idx,
-                             " data_dim=", input_data_shape[axis]);
+    Tin idx = indices_data_raw[i];
+
+    if ((idx >= axis_dim_limit) ||  
+        // 'Scatter' does not support negative indices
+        (idx < 0 && !is_scatter_elements) ||
```

Comment:
Whilst the spec doesn't explicitly support it, there's no reason why we can't gracefully handle it. We did that for a number of other ops that didn't explicitly say negative axis values were supported.

Reply:
Well the previous implementation did not support it and it seems unnecessary to enhance it when the spec is unclear. I am open to supporting it for the previous version as well - but this would be enhancing the older version while this change keeps it consistent with the previous version...

Reply:
Fixed